File: E-Tour/Java/Backend/E-Tour/src/main/java/com/example/models/Customer.java
package com.example.models;

import java.util.List;

import jakarta.persistence.*;

@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="customerid")
    private Long Customerid;

    @Column(name="email")
    private String email;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false)
    private String password;

    private Long phoneNumber1;
    
    
    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Passenger> passengers;  // One-to-Many relationship
    
    public List<Passenger> getPassengers() {
        return passengers;
    }

    public void setPassengers(List<Passenger> passengers) {
        this.passengers = passengers;
    }

    // Default Constructor
    public Customer() {
    }

    // Getters and Setters
    public Long getId() {
        return Customerid;
    }

    public void setId(Long id) {
        this.Customerid = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Long getPhoneNumber1() {
        return phoneNumber1;
    }

    public void setPhoneNumber1(Long phoneNumber1) {
        this.phoneNumber1 = phoneNumber1;
    }


	@Override
	public String toString() {
		return "Customer [email=" + email + ", password=" + password + "]";
	}
    
    
}

File: Models/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace E_Tour.Models;

[Table("customer")]
public partial class Customer
{
    [Key]
    [Column("customerid")]
    public long Customerid { get; set; }

    [Column("email")]
    [StringLength(255)]
    public string? email { get; set; }

    [Column("firstName")]
    [StringLength(255)]
    public string firstName { get; set; } = null!;

    [Column("lastName")]
    [StringLength(255)]
    public string lastName { get; set; } = null!;

    [Column("password")]
    [StringLength(255)]
    public string password { get; set; } = null!;

    [Column("phoneNumber1")]
    public long? phoneNumber1 { get; set; } 

    [InverseProperty("Customer")]
    public virtual ICollection<Passenger> Passengers { get; set; } = new List<Passenger>();
}
